pkg/ccl/changefeedccl/enriched_source_provider.go

```diff
@@ -163,9 +150,115 @@ func (p *enrichedSourceProvider) GetJSON(updated cdcevent.Row) (json.JSON, error
 func (p *enrichedSourceProvider) GetAvro(
 	row cdcevent.Row, schemaPrefix string,
 ) (*avro.FunctionalRecord, error) {
-	sourceDataSchema, err := avro.NewFunctionalRecord("source", schemaPrefix, p.getAvroFields(), p.avroSourceFunction)
+	sourceDataSchema, err := avro.NewFunctionalRecord("source", schemaPrefix, avroFields, p.avroSourceFunction)
 	if err != nil {
 		return nil, err
 	}
 	return sourceDataSchema, nil
 }
+
+type fieldInfo struct {
+	avroSchemaField    avro.SchemaField
+	kafkaConnectSchema kcjsonschema.Schema
+}
+
+var allFieldInfo = map[string]fieldInfo{
+	"job_id": {
+		avroSchemaField: avro.SchemaField{
+			Name:       "job_id",
+			SchemaType: []avro.SchemaType{avro.SchemaTypeNull, avro.SchemaTypeString},
+		},
+		kafkaConnectSchema: kcjsonschema.Schema{
+			Field:    "job_id",
+			TypeName: kcjsonschema.SchemaTypeString,
+			Optional: true,
+		},
+	},
+	"db_version": {
+		avroSchemaField: avro.SchemaField{
+			Name:       "db_version",
+			SchemaType: []avro.SchemaType{avro.SchemaTypeNull, avro.SchemaTypeString},
+		},
+		kafkaConnectSchema: kcjsonschema.Schema{
+			Field:    "db_version",
+			TypeName: kcjsonschema.SchemaTypeString,
+			Optional: true,
+		},
+	},
+	"cluster_name": {
+		avroSchemaField: avro.SchemaField{
+			Name:       "cluster_name",
+			SchemaType: []avro.SchemaType{avro.SchemaTypeNull, avro.SchemaTypeString},
+		},
+		kafkaConnectSchema: kcjsonschema.Schema{
+			Field:    "cluster_name",
+			TypeName: kcjsonschema.SchemaTypeString,
+			Optional: true,
+		},
+	},
+	"cluster_id": {
+		avroSchemaField: avro.SchemaField{
+			Name:       "cluster_id",
+			SchemaType: []avro.SchemaType{avro.SchemaTypeNull, avro.SchemaTypeString},
+		},
+		kafkaConnectSchema: kcjsonschema.Schema{
+			Field:    "cluster_id",
+			TypeName: kcjsonschema.SchemaTypeString,
+			Optional: true,
+		},
+	},
+	"source_node_locality": {
+		avroSchemaField: avro.SchemaField{
+			Name:       "source_node_locality",
+			SchemaType: []avro.SchemaType{avro.SchemaTypeNull, avro.SchemaTypeString},
+		},
+		kafkaConnectSchema: kcjsonschema.Schema{
+			Field:    "source_node_locality",
+			TypeName: kcjsonschema.SchemaTypeString,
+			Optional: true,
+		},
+	},
+	"node_name": {
+		avroSchemaField: avro.SchemaField{
+			Name:       "node_name",
+			SchemaType: []avro.SchemaType{avro.SchemaTypeNull, avro.SchemaTypeString},
+		},
+		kafkaConnectSchema: kcjsonschema.Schema{
+			Field:    "node_name",
+			TypeName: kcjsonschema.SchemaTypeString,
+			Optional: true,
+		},
+	},
+	"node_id": {
+		avroSchemaField: avro.SchemaField{
+			Name:       "node_id",
+			SchemaType: []avro.SchemaType{avro.SchemaTypeNull, avro.SchemaTypeString},
+		},
+		kafkaConnectSchema: kcjsonschema.Schema{
+			Field:    "node_id",
+			TypeName: kcjsonschema.SchemaTypeString,
+			Optional: true,
+		},
+	},
+}
+
+// filled in by init() using allFieldInfo
+var avroFields []*avro.SchemaField
+
+// filled in by init() using allFieldInfo
+var kcjSchema kcjsonschema.Schema
```

Comment:
nit: would prefer to see a longer and more descriptive name like `kafkaConnectJSONSchema`